code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -247,6 +277,26 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
 		qdel(loc)
 	qdel(src)
 
+/obj/structure/telecomms_shield_cover
+	name = "turret"
+	desc = "Looks like the cover to a turret. Not deploying, however?"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	anchored = TRUE
+
+/obj/structure/telecomms_shield_cover/proc/activate()
+	invisibility = 90
+	var/obj/machinery/shieldgen/telecomms/trap = new /obj/machinery/shieldgen/telecomms(get_turf(src))
+	var/atom/flick_holder = new /atom/movable/porta_turret_cover(loc)
+	flick_holder.layer = trap.layer + 0.1
+	flick("popup", flick_holder)
```

Comment:
Didn't flick have issues sometimes?

Reply:
It does related to how often flick is played. Ideally it was fully purged, yes, but something that flicks once is much less of a problem.

Reply:
Sounds good then